app/models/template.rb

```diff
@@ -200,6 +200,10 @@ def sync_inputs(inputs)
     inputs.each_value { |new_input| template_inputs.build(new_input) }
   end
 
+  def support_host_based_preview?
```

Comment:
The name could be bit more generic, not only the preview is host based, but every render.
I'm ok with that for now tho, we can rename once we find another usecase for it.